python_modules/libraries/dagster-looker/dagster_looker/api/resource.py

```diff
@@ -126,16 +140,52 @@ def _build_defs_from_looker_instance_data(
             )
             for lookml_explore in looker_instance_data.explores_by_id.values()
         ]
-        views = [
+        dashboards = [
             dagster_looker_translator.get_asset_spec(
                 LookerStructureData(
                     structure_type=LookerStructureType.DASHBOARD, data=looker_dashboard
                 )
             )
             for looker_dashboard in looker_instance_data.dashboards_by_id.values()
         ]
+        dashboards_postprocessed = self._build_dashboard_notif_defs(
+            scheduled_plans_by_dashboard_id,
+            list(looker_instance_data.dashboards_by_id.values()),
+            dashboards,
+        )
+
+        return Definitions(assets=[*pdts, *explores, *dashboards_postprocessed])
+
+    def _build_dashboard_notif_defs(
+        self,
+        destinations: Dict[str, "ScheduledPlanDestination"],
+        dashboards: Sequence["Dashboard"],
+        specs: Sequence[AssetSpec],
+    ) -> Sequence[Union[AssetsDefinition, AssetSpec]]:
```

Comment:
Could we return a list of `AssetsDefinitions` and another of `AssetsSpecs`? The implementation makes sense to me - it was just not obvious at first that we were returning asset specs for the dashboards that have no `ScheduledPlanDestination`.

Reply:
Updated to use builder pattern, which I think makes this more clear